<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>QgsTableEditorFormattingWidgetBase</class>
 <widget class="QWidget" name="QgsTableEditorFormattingWidgetBase">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>410</width>
    <height>464</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Attribute Table</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <property name="spacing">
    <number>0</number>
   </property>
   <property name="leftMargin">
    <number>0</number>
   </property>
   <property name="topMargin">
    <number>0</number>
   </property>
   <property name="rightMargin">
    <number>0</number>
   </property>
   <property name="bottomMargin">
    <number>0</number>
   </property>
   <item>
    <widget class="QgsScrollArea" name="scrollArea">
     <property name="frameShape">
      <enum>QFrame::NoFrame</enum>
     </property>
     <property name="widgetResizable">
      <bool>true</bool>
     </property>
     <widget class="QWidget" name="scrollAreaWidgetContents">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>0</y>
        <width>410</width>
        <height>464</height>
       </rect>
      </property>
      <layout class="QGridLayout" name="gridLayout">
       <item row="5" column="0">
        <spacer name="verticalSpacer">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>0</height>
          </size>
         </property>
        </spacer>
       </item>
       <item row="0" column="0">
        <widget class="QLabel" name="label_5">
         <property name="text">
          <string>Expression</string>
         </property>
        </widget>
       </item>
       <item row="2" column="0" colspan="2">
        <widget class="QGroupBox" name="groupBox_6">
         <property name="focusPolicy">
          <enum>Qt::StrongFocus</enum>
         </property>
         <property name="title">
          <string>Colors</string>
         </property>
         <property name="syncGroup" stdset="0">
          <string notr="true">composeritem</string>
         </property>
         <property name="collapsed" stdset="0">
          <bool>false</bool>
         </property>
         <layout class="QGridLayout" name="gridLayout_6" columnstretch="2,2,1">
          <item row="3" column="1">
           <widget class="QgsColorButton" name="mBackgroundColorButton">
            <property name="minimumSize">
             <size>
              <width>120</width>
              <height>0</height>
             </size>
            </property>
            <property name="maximumSize">
             <size>
              <width>120</width>
              <height>16777215</height>
             </size>
            </property>
            <property name="text">
             <string/>
            </property>
           </widget>
          </item>
          <item row="2" column="0">
           <widget class="QLabel" name="label_10">
            <property name="text">
             <string>Text color</string>
            </property>
           </widget>
          </item>
          <item row="3" column="0">
           <widget class="QLabel" name="label_11">
            <property name="text">
             <string>Background color</string>
            </property>
           </widget>
          </item>
          <item row="2" column="1">
           <widget class="QgsColorButton" name="mTextColorButton">
            <property name="minimumSize">
             <size>
              <width>120</width>
              <height>0</height>
             </size>
            </property>
            <property name="maximumSize">
             <size>
              <width>120</width>
              <height>16777215</height>
             </size>
            </property>
            <property name="text">
             <string/>
            </property>
           </widget>
          </item>
          <item row="2" column="2">
           <spacer name="horizontalSpacer">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </widget>
       </item>
       <item row="3" column="0" colspan="2">
        <widget class="QGroupBox" name="groupBox_7">
         <property name="focusPolicy">
          <enum>Qt::StrongFocus</enum>
         </property>
         <property name="title">
          <string>Formatting</string>
         </property>
         <property name="syncGroup" stdset="0">
          <string notr="true">composeritem</string>
         </property>
         <property name="collapsed" stdset="0">
          <bool>false</bool>
         </property>
         <layout class="QGridLayout" name="gridLayout_7">
          <item row="1" column="0">
           <widget class="QCheckBox" name="mFormatNumbersCheckBox">
            <property name="text">
             <string>Format as number</string>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QPushButton" name="mCustomizeFormatButton">
            <property name="text">
             <string>Customizeâ€¦</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QgsFontButton" name="mFontButton">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="text">
             <string>Font</string>
            </property>
           </widget>
          </item>
          <item row="2" column="0">
           <widget class="QLabel" name="label_3">
            <property name="text">
             <string>Horizontal alignment</string>
            </property>
           </widget>
          </item>
          <item row="3" column="0">
           <widget class="QLabel" name="label_4">
            <property name="text">
             <string>Vertical alignment</string>
            </property>
           </widget>
          </item>
          <item row="3" column="1">
           <widget class="QgsAlignmentComboBox" name="mVerticalAlignComboBox"/>
          </item>
          <item row="2" column="1">
           <widget class="QgsAlignmentComboBox" name="mHorizontalAlignComboBox"/>
          </item>
          <item row="0" column="0">
           <widget class="QLabel" name="label_5">
            <property name="text">
             <string>Text format</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item row="4" column="0" colspan="2">
        <widget class="QGroupBox" name="groupBox_8">
         <property name="focusPolicy">
          <enum>Qt::StrongFocus</enum>
         </property>
         <property name="title">
          <string>Cell Size</string>
         </property>
         <property name="syncGroup" stdset="0">
          <string notr="true">composeritem</string>
         </property>
         <property name="collapsed" stdset="0">
          <bool>false</bool>
         </property>
         <layout class="QGridLayout" name="gridLayout_8" columnstretch="0,0">
          <item row="0" column="1">
           <widget class="QgsDoubleSpinBox" name="mRowHeightSpinBox">
            <property name="suffix">
             <string> mm</string>
            </property>
            <property name="maximum">
             <double>999999.000000000000000</double>
            </property>
           </widget>
          </item>
          <item row="0" column="0">
           <widget class="QLabel" name="label">
            <property name="text">
             <string>Row height</string>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QgsDoubleSpinBox" name="mColumnWidthSpinBox">
            <property name="suffix">
             <string> mm</string>
            </property>
            <property name="maximum">
             <double>999999.000000000000000</double>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QLabel" name="label_2">
            <property name="text">
             <string>Column width</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item row="0" column="1">
        <widget class="QgsFieldExpressionWidget" name="mExpressionEdit" native="true">
         <property name="sizePolicy">
          <sizepolicy hsizetype="MinimumExpanding" vsizetype="Preferred">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="focusPolicy">
          <enum>Qt::StrongFocus</enum>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>QgsDoubleSpinBox</class>
   <extends>QDoubleSpinBox</extends>
   <header>qgsdoublespinbox.h</header>
  </customwidget>
  <customwidget>
   <class>QgsColorButton</class>
   <extends>QToolButton</extends>
   <header>qgscolorbutton.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>QgsScrollArea</class>
   <extends>QScrollArea</extends>
   <header>qgsscrollarea.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>QgsFontButton</class>
   <extends>QToolButton</extends>
   <header>qgsfontbutton.h</header>
  </customwidget>
  <customwidget>
   <class>QgsAlignmentComboBox</class>
   <extends>QComboBox</extends>
   <header>qgsalignmentcombobox.h</header>
  </customwidget>
  <customwidget>
   <class>QgsFieldExpressionWidget</class>
   <extends>QWidget</extends>
   <header>qgsfieldexpressionwidget.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <tabstops>
  <tabstop>groupBox_6</tabstop>
  <tabstop>mTextColorButton</tabstop>
  <tabstop>mBackgroundColorButton</tabstop>
  <tabstop>groupBox_7</tabstop>
  <tabstop>mFontButton</tabstop>
  <tabstop>mFormatNumbersCheckBox</tabstop>
  <tabstop>mCustomizeFormatButton</tabstop>
  <tabstop>mHorizontalAlignComboBox</tabstop>
  <tabstop>mVerticalAlignComboBox</tabstop>
  <tabstop>groupBox_8</tabstop>
  <tabstop>mRowHeightSpinBox</tabstop>
  <tabstop>mColumnWidthSpinBox</tabstop>
  <tabstop>scrollArea</tabstop>
 </tabstops>
 <resources/>
 <connections/>
</ui>
